crates/voicevox_core/src/user_dict/dict.rs

```diff
@@ -0,0 +1,85 @@
+use derive_getters::Getters;
+use fs_err::File;
+use indexmap::IndexMap;
+use itertools::join;
+use uuid::Uuid;
+
+use super::word::*;
+use crate::{Error, Result};
+
+/// ユーザー辞書。
+/// 単語はJSONとの相互変換のために挿入された順序を保つ。
+#[derive(Clone, Debug, Default, Getters)]
+pub struct UserDict {
+    words: IndexMap<Uuid, UserDictWord>,
+}
+
+impl UserDict {
+    /// ユーザー辞書を作成する。
+    pub fn new() -> Self {
+        Default::default()
+    }
+
+    /// ユーザー辞書をファイルから読み込む。
+    ///
+    /// ファイルが存在しない、または内容が不正の場合はエラーを返す。
+    pub fn load(&mut self, store_path: &str) -> Result<()> {
+        let store_path = std::path::Path::new(store_path);
+        if !store_path.exists() {
+            return Err(Error::LoadUserDict("ファイルが存在しません".to_string()));
+        }
```

Comment:
<https://github.com/VOICEVOX/voicevox_core/pull/538#discussion_r1263992168>の意図としては、no such fileのエラーもfs-errに扱わせたらよいのではないかというものでした。

```suggestion
    /// ファイルが読めなかった、または内容が不正だった場合はエラーを返す。
    pub fn load(&mut self, store_path: &str) -> Result<()> {
        let store_path = std::path::Path::new(store_path);
```

Reply:
なるほど。